block/manager.go

```diff
@@ -250,6 +250,7 @@ func (m *Manager) SetFraudProofService(fraudProofServ *fraudserv.ProofService) {
 }
 
 func (m *Manager) ProcessFraudProof(ctx context.Context, cancel context.CancelFunc) {
+	defer cancel()
```

Comment:
Why is this method cancelling the inbound context?  

I would think that either the caller should handle the context cancel.

This might be a longer conversation about this method that we can handle in a follow up, but the update to the defer cancel is a good step over the cancel at the end of the method.

Reply:
I don't have the context for this. Why was `ProcessFraudProof` written this way? @gupadhyaya

Reply:
`ProcessFraudProof` will block when there is no valid fraud proof. If it finds a valid fraud proof, the current behavior is halting which is done via context cancel. The issue https://github.com/rollkit/rollkit/issues/933#issuecomment-1568823982 will address further on how to gracefully halt on fraud.